src/include/sof/math/trig.h

```diff
@@ -137,17 +144,189 @@ static inline int16_t sin_fixed_16b(int32_t th_rad_fxp)
+ * | thRadFxp	      |cdccexpth        |thRadFxp|   cdccexpth|
+ * +----+-----+-------+----+----+-------+--------+------------+
+ * |WLen| FLen|Signbit|WLen|FLen|Signbit| Qformat| Qformat    |
+ * +----+-----+-------+----+----+-------+--------+------------+
+ * | 32 | 28  |  1    | 32 | 15 |   1	| 4.28	 | 1.15       |
+ * +------------------+-----------------+--------+------------+
+ */
+static inline void cmpx_exp_16b(int32_t th_rad_fxp, struct cordic_cmpx *cexp)
+{
+	int32_t th_cdc_fxp;
+	int32_t sign;
+	int32_t b_yn;
+	int32_t xn;
+
+	/* compute coeff from angles */
+	cordic_approx(th_rad_fxp, CORDIC_15B_TABLE_SIZE, &sign, &b_yn, &xn, &th_cdc_fxp);
+	cmpx_cexp(sign, b_yn, xn, EN_16B_CORDIC_CEXP, cexp);
+	/* return the complex(re & im) result in Q1.15*/
+}
+
+/**
+ * CORDIC-based approximation of inverse sine
+ * inverse sine of cdc_asin_th based on a CORDIC approximation.
+ * Inverse sine angle values in rad
+ * Q2.30 cdc_asin_th, value in between range of [-1 to 1]
+ * Q2.30 th_rad_fxp, output value range [-1.5707963258028 to 1.5707963258028]
+ * LUT size set type 15
+ * Error (max = 0.000000027939677), THD+N  = -157.454534077921551 (dBc)
+ */
+static inline int32_t asin_fixed_32b(int32_t saturated_unary_minus)
+{
+	int32_t th_rad_fxp;
+
+	if (saturated_unary_minus >= 0) {
+		th_rad_fxp = is_scalar_cordic_asin(saturated_unary_minus,
+						   CORDIC_31B_TABLE_SIZE,
+						   cordic_ilookup);
+	} else {
+		th_rad_fxp = -is_scalar_cordic_asin(-saturated_unary_minus,
+						    CORDIC_31B_TABLE_SIZE,
+						    cordic_ilookup);
+	}
+	return th_rad_fxp; /* Q2.30 */
+}
+
+/**
+ * CORDIC-based approximation of inverse cosine
+ * inverse cosine of cdc_acos_th based on a CORDIC approximation
+ * acos(X) inverse cosine angle values in radian produces using the DCORDIC
+ * (Double CORDIC) algorithm.
+ * Q2.30 cdc_acos_th , input value range [-1 to 1]
+ * Q3.29 th_rad_fxp, output value range [3.14159265346825 to 0]
+ * LUT size set type 31
+ * Error (max = 0.000000026077032), THD+N  = -157.948952635422842 (dBc)
+ */
+static inline int32_t acosine_fixed_32b(int32_t saturated_unary_minus)
+{
+	int32_t th_rad_fxp;
+
+	if (saturated_unary_minus >= 0) {
+		th_rad_fxp = is_scalar_cordic_acos(saturated_unary_minus,
+						   CORDIC_31B_TABLE_SIZE,
+						   cordic_ilookup);
+	} else {
+		th_rad_fxp =
+		PI_MUL2_Q4_28 - is_scalar_cordic_acos(-saturated_unary_minus,
+						      CORDIC_31B_TABLE_SIZE,
+						      cordic_ilookup);
+	}
+	return th_rad_fxp; /* Q3.29 */
+}
+
+/**
+ * CORDIC-based approximation of inverse sine
+ * inverse sine of cdc_asin_th based on a CORDIC approximation.
+ * Inverse sine angle values in rad
+ * Q2.30 cdc_asin_th, value in between range of [-1 to 1]
+ * Q1.15 th_rad_fxp, output value range [-1.5707963258028 to 1.5707963258028]
+ * LUT size set type 31
+ * number of iteration 15
+ * Error (max = 0.000059800222516), THD+N  = -89.824282520774048 (dBc)
+ */
+static inline int32_t asin_fixed_16b(int32_t saturated_unary_minus)
```

Comment:
I think it would be better to return int16_t instead of int32_t
1. aligned to sin_fixed_16b and cos_fixed_16b
2. sat_int16(*) generates int16_t

Reply:
Yes, good observation, it is better to return int16_t rather int32_t.